src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -358,7 +360,10 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
     s.asInstanceOf[S with EfficientSplit]
   }
 
-  private[this] def growTable(newlen: Int) = {
+  private[this] def growTable(newlenLong: Long) = {
```

Comment:
we could probably just leave it as an `Int` and check if it is negative (and assume it has overflowed). if this were being called for every action, I would be concerned about potentially unnecessary conversions, but as it is only called when growing the table, and as that action is already quite expensive, I don't think it has a meaningful performance cost. I'm happy with either the way it is, or checking if it's a negative `Int`; whichever folks think is clearer.

Reply:
I followed your recommendation and switched back to `Int`. Thanks